### PR TITLE
CTCP-3665: Migration from Hardcoded codelists to Reference data for DeclarationType

### PR DESCRIPTION
**Pull request by joecefnhedog**

### CODE REVIEW

File: app/services/DeclarationTypeService.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  private def sort(declarationType: Seq[DeclarationType]): Seq[DeclarationType] =
+    declarationType.sortBy(_.toString.toLowerCase)
+
+}
+
+object DeclarationTypeService {
+
+  def filterUserAnswers(declarationType: Seq[DeclarationType]): Seq[DeclarationType] =
+    declarationType.filterNot(
+      decType => decType.code == T2F
```
Comment (joecefnhedog) [inline app/services/DeclarationTypeService.scala:47]: We want to filter this out from reference data, clarified with Sayak: https://jira.tools.tax.service.gov.uk/browse/CTCP-3665

File: app/services/DeclarationTypeService.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  private def sort(declarationType: Seq[DeclarationType]): Seq[DeclarationType] =
+    declarationType.sortBy(_.toString.toLowerCase)
+
+}
+
+object DeclarationTypeService {
+
+  def filterUserAnswers(declarationType: Seq[DeclarationType]): Seq[DeclarationType] =
+    declarationType.filterNot(
+      decType => decType.code == T2SM
```
Comment (joecefnhedog) [inline app/services/DeclarationTypeService.scala:47]: We want to filter this out from reference data, clarified with Sayak: https://jira.tools.tax.service.gov.uk/browse/CTCP-3665

**Comment by jackheslop96** [comment]
@joecefnhedog is there anything in here which isn't in https://github.com/hmrc/manage-transit-movements-departure-frontend/pull/567? Just trying to get a gras of what we need to merge when we can start merging again

**Comment by joecefnhedog** [comment]
> @joecefnhedog is there anything in here which isn't in #567? Just trying to get a gras of what we need to merge when we can start merging again

I believe this is duped work, it can be closed.

**Comment by joecefnhedog** [comment]
Work was completed in 3661

**Pull request closed without merging**